p2p/src/msg.rs

```diff
@@ -213,6 +213,92 @@ pub fn read_header(conn: &mut TcpStream, msg_type: Option<Type>) -> Result<MsgHe
 	Ok(header)
 }
 
+/// A reader that reads straight off a stream.
```

Comment:
Considered putting that in core `ser.rs`? Seems that it could be reused in other places over time.

Reply:
Nevermind, hadn't realized it relied on our `read_exact` definition.

Reply:
Done.
It has a dependency on our custom `read_exact()` so moved that out of `p2p` and into `util`. This now uses a generic `Read` and not a specific `TcpStream`.